### DISCUSSION TITLE
Using Transactions with unique and exists rules

### DISCUSSION THREAD

**Discussion by mastermunj** [adonisjs/core#1912] [Help] [answered]
Hi @thetutlage,

Is there a way transactions can be used for unique and exists validators?

The Use case:
* We store seed data in json files named after Model names.
* We have created a command that accepts name of Model as parameter and
  * Instantiates Repository (which is responsible to validate data before calling Model.create)
  * Drop Table
  * Loop through data imported from json, validate and create new rows.

What we want is to do all of these within a transaction to ensure if there are any validation issues, database table remains in the previous state. Transactions works very well with delete and create, however challenge comes with validations.

Since validators doesn't support transactions, the queries from validator runs on current live data on table rather than data within transaction.

**Comment by thetutlage** [comment, marked as answer]
How about using global transactions? 

```ts
import Database from '@ioc:Adonis/Core/Database'

await Database.beginGlobalTranaction()

// seed data

await Database.commitGlobalTransaction()
```

**Reply by mastermunj** [reply]
Works like charm! Will enabling global transaction have any side effect on any other database queries happening in parallel? Or this is effective only for the current running command?

**Reply by thetutlage** [reply]
It will effect parallel queries only if they are part of the same process. But I believe, the command to seed data does only one thing that is seed data

**Reply by mastermunj** [reply]
Correct. Seed data only does one job.
